api/services/StatisticsMassifService.js

```diff
@@ -0,0 +1,231 @@
+const CREATE_MATERIALIZED_VIEW = `
+  CREATE MATERIALIZED VIEW v_massif_info AS
+  SELECT m.id as id_massif,
+  c.id as id_cave,
+  n.name as name_cave,
+  c.depth as depth_cave,
+  c.length as length_cave,
+  c.is_diving as is_diving_cave,
+  COUNT(DISTINCT e.id) as nb_entrances
+  FROM t_massif m 
+  JOIN t_entrance e ON ST_Contains(ST_SetSRID(m.geog_polygon::geometry, 4326), ST_SetSRID(ST_MakePoint(e.longitude, e.latitude), 4326))
+  JOIN t_cave c ON e.id_cave = c.id 
+  JOIN t_name n ON n.id_cave = c.id 
+  WHERE n.is_main = true 
+  AND c.is_deleted = false
+  AND e.is_deleted = false
+  AND m.is_deleted = false
+  GROUP BY(m.id, c.id, n.name, c.depth, c.length, c.is_diving );
+`;
+
+const GET_NB_CAVES = `
+  SELECT COUNT(*) as nb_caves
+  FROM v_massif_info
+  WHERE id_massif = $1
+`;
+
+const GET_NB_NETWORKS = `
+  SELECT COUNT(*) as nb_networks
+  FROM v_massif_info
+  WHERE id_massif = $1
+  AND nb_entrances > 1
+`;
+
+const FIND_CAVE_WITH_MAX_DEPTH_IN_MASSIF = `
+  SELECT id_cave, name_cave, depth_cave
+  FROM v_massif_info
+  WHERE depth_cave IN (
+    SELECT MAX(depth_cave)
+    FROM v_massif_info
+    WHERE id_massif = $1
+    )
+  AND id_massif = $1
+`;
+
+const FIND_CAVE_WITH_MAX_LENGTH_IN_MASSIF = `
+  SELECT id_cave, name_cave, length_cave
+  FROM v_massif_info
+  WHERE length_cave IN (
+    SELECT MAX(length_cave)
+    FROM v_massif_info
+    WHERE id_massif = $1
+  )
+  AND id_massif = $1
+  LIMIT 1
+`;
+
+const GET_NB_CAVES_WHICH_ARE_DIVING_IN_MASSIF = `
+  SELECT COUNT(*) as nb_diving_cave
+  FROM v_massif_info
+  WHERE is_diving_cave = true
+  AND id_massif = $1
+`;
+
+const GET_AVG_DEPTH_AND_LENGTH_IN_MASSIF = `
+  SELECT AVG(depth_cave) as avg_depth, AVG(length_cave) as avg_length
+  FROM v_massif_info
+  WHERE id_massif = $1 
+`;
+
+const GET_TOTAL_LENGTH_IN_MASSIF = `
+  SELECT SUM(length_cave) as value, COUNT(length_cave) as nb_data
+  FROM v_massif_info
+  WHERE length_cave IS NOT NULL
+  AND id_massif = $1
+`;
+
+const CommonService = require('./CommonService');
+
+module.exports = {
+  createMaterializedView: async () => {
+    try {
+      const queryResult = await CommonService.query(CREATE_MATERIALIZED_VIEW);
+      return queryResult.rows;
+    } catch (e) {
+      return [];
+    }
+  },
+
+  /**
+   *
+   * @param {int} massifId
+   * @returns {int} the number of caves in the massif
+   */
+  getNbCavesInMassif: async (massifId) => {
+    try {
+      const queryResult = await CommonService.query(GET_NB_CAVES, [massifId]);
+      const result = queryResult.rows;
+      if (result.length > 0) {
+        return result[0];
+      }
+      return null;
```

Comment:
Do you want to return null or 0 here ?

Reply:
We want `null` indeed because 0 means that there is no entity researched (a cave, network, entrance,...) and null means a lack of data (as a massif which does not exist or is not having a geog polygon filled in the DB).

Reply:
ok, so the `null` case will need to be handled differently than the 0 case by the front end :)
maybe it should be documented in the source code here ?